apps/status-api/__test__/controllers/OverallStatusController.test.ts

```diff
@@ -0,0 +1,118 @@
+import { StatusApiTesting } from '../../testing/StatusApiTesting'
+import { ApiPagedResponse, WhaleApiClient } from '@defichain/whale-api-client'
+import { WhaleApiProbeIndicator } from '../../src/modules/WhaleApiModule'
+import { HealthIndicatorResult, HealthIndicatorStatus } from '@nestjs/terminus'
+
+describe('AggregateController - Status test', () => {
+  const apiTesting = StatusApiTesting.create()
+  beforeAll(async () => {
+    await apiTesting.start()
+  })
+
+  afterAll(async () => {
+    await apiTesting.stop()
+  })
+
+  it('/overall - should get operational when Ocean and blockchain are up', async () => {
+    jest
+      .spyOn(apiTesting.app.get(WhaleApiProbeIndicator), 'liveness')
+      .mockReturnValueOnce(getWhaleStatus('up'))
+
+    jest
+      .spyOn(apiTesting.app.get(WhaleApiClient).blocks, 'list')
+      .mockReturnValueOnce(getBlockResponseWithPresetTime(25))
+
+    const res = await apiTesting.app.inject({
+      method: 'GET',
+      url: '/overall'
+    })
+
+    expect(res.statusCode).toStrictEqual(200)
+    expect(res.json()).toStrictEqual({
+      status: 'operational'
+    })
+  })
+
+  it('/overall - should get outage when blockchain is down', async () => {
+    jest
+      .spyOn(apiTesting.app.get(WhaleApiClient).blocks, 'list')
+      .mockReturnValueOnce(getBlockResponseWithPresetTime(46))
+
+    const res = await apiTesting.app.inject({
+      method: 'GET',
+      url: '/overall'
+    })
+
+    expect(res.statusCode).toStrictEqual(200)
+    expect(res.json()).toStrictEqual({
+      status: 'outage'
+    })
+  })
+
+  it('/overall - should get degraded when blockchain degraded', async () => {
+    jest
+      .spyOn(apiTesting.app.get(WhaleApiClient).blocks, 'list')
+      .mockReturnValueOnce(getBlockResponseWithPresetTime(36))
+
+    const res = await apiTesting.app.inject({
+      method: 'GET',
+      url: '/overall'
+    })
+
+    expect(res.statusCode).toStrictEqual(200)
+    expect(res.json()).toStrictEqual({
+      status: 'degraded'
+    })
+  })
+
+  it('/overall - should get outage when Ocean down', async () => {
+    jest
+      .spyOn(apiTesting.app.get(WhaleApiProbeIndicator), 'liveness')
+      .mockReturnValueOnce(getWhaleStatus('down'))
+
+    const res = await apiTesting.app.inject({
+      method: 'GET',
+      url: '/overall'
+    })
+
+    expect(res.statusCode).toStrictEqual(200)
+    expect(res.json()).toStrictEqual({
+      status: 'outage'
+    })
+  })
+})
+
+async function getWhaleStatus (status: HealthIndicatorStatus): Promise<HealthIndicatorResult> {
+  return {
+    whale: {
+      status
+    }
+  }
+}
+
+async function getBlockResponseWithPresetTime (minutesDiff: number): Promise<ApiPagedResponse<any>> {
```

Comment:
Typescript hates any
```suggestion
async function getBlockResponseWithPresetTime (minutesDiff: number): Promise<ApiPagedResponse<Block>> {
```
Block should get from 
`import { Block } from '@defichain/whale-api-client/dist/api/Blocks'`

Apply to other same code in the project

Reply:
Nice catch there.